src/emitter/batch_emitter.rs

```diff
@@ -179,10 +182,12 @@ impl BatchEmitter {
 
                         // Spawn a new task to send the batch
                         tokio_tasks.push(tokio::spawn(async move {
-                            // TODO: Change println to logging
+                            let batch_length = batch.events.len();
                             match Self::send_batch(batch, client).await {
-                                Ok(_) => println!("Batch sent"),
-                                Err(e) => println!("Error sending batch: {e}"),
+                                Ok(_) => {
+                                    log::debug!("Sent batch of {batch_length} events")
```

Comment:
Nit: wondering if this would better be an info-level log? It seems a useful information to know when events were sent to the collector even if not debugging the tracking.